packages/flutter/lib/src/material/time.dart

```diff
@@ -112,6 +112,30 @@ class TimeOfDay {
     );
   }
 
+  /// Whether this [TimeOfDay] occurs earlier than [other].
+  bool isBefore(TimeOfDay other) => compareTo(other) < 0;
+
+  /// Whether this [TimeOfDay] occurs later than [other].
+  bool isAfter(TimeOfDay other) => compareTo(other) > 0;
+
+  /// Whether this [TimeOfDay] occurs at the same time [other].
+  bool isAtSameTimeAs(TimeOfDay other) => compareTo(other) == 0;
+
+  /// Compares this [TimeOfDay] object to [other] independent of date.
+  ///
+  /// This means that "00:00" of the next day is still before "23:00" of this
+  /// day.
+  ///
+  /// A [compareTo] function returns:
+  ///  * a negative value if this TimeOfDay [isBefore] [other].
+  ///  * `0` if this DateTime [isAtSameTimeAs] [other], and
+  ///  * a positive value otherwise (when this TimeOfDay [isAfter] [other]).
+  @override
+  int compareTo(TimeOfDay other) {
+    final int hourComparison = hour.compareTo(other.hour);
+    return hourComparison == 0 ? minute.compareTo(other.minute) : hourComparison;
```

Comment:
Here's a suggestion to make the comparison slightly more intuitive:

```suggestion
    return 60 * hour.compareTo(other.hour) + minute.compareTo(other.minute)
```

Reply:
I don't think `compareTo` makes more guarantees than less than or greater than zero. While this would probably work (both are integer comparisons) I think this misuses the method.

Reply:
From a quick benchmark it also appears like your intuitive implementation is slightly slower:

```
Conditional compare(RunTime): 74305.60714285714 us.
Math compare(RunTime): 84825.04166666667 us.
```

<details><summary>Code:</summary>
<p>

```dart
import 'package:benchmark_harness/benchmark_harness.dart';

class TimeOfDay {
  /// Creates a time of day.
  ///
  /// The [hour] argument must be between 0 and 23, inclusive. The [minute]
  /// argument must be between 0 and 59, inclusive.
  const TimeOfDay({ required this.hour, required this.minute });

  /// Creates a time of day based on the given time.
  ///
  /// The [hour] is set to the time's hour and the [minute] is set to the time's
  /// minute in the timezone of the given [DateTime].
  TimeOfDay.fromDateTime(DateTime time)
      : hour = time.hour,
        minute = time.minute;

  /// Creates a time of day based on the current time.
  ///
  /// The [hour] is set to the current hour and the [minute] is set to the
  /// current minute in the local time zone.
  TimeOfDay.now() : this.fromDateTime(DateTime.now());

  /// The number of hours in one day, i.e. 24.
  static const int hoursPerDay = 24;

  /// The number of hours in one day period (see also [DayPeriod]), i.e. 12.
  static const int hoursPerPeriod = 12;

  /// The number of minutes in one hour, i.e. 60.
  static const int minutesPerHour = 60;

  /// Returns a new TimeOfDay with the hour and/or minute replaced.
  TimeOfDay replacing({ int? hour, int? minute }) {
    assert(hour == null || (hour >= 0 && hour < hoursPerDay));
    assert(minute == null || (minute >= 0 && minute < minutesPerHour));
    return TimeOfDay(hour: hour ?? this.hour, minute: minute ?? this.minute);
  }

  /// The selected hour, in 24 hour time from 0..23.
  final int hour;

  /// The selected minute.
  final int minute;


  @override
  bool operator ==(Object other) {
    return other is TimeOfDay
        && other.hour == hour
        && other.minute == minute;
  }

  @override
  int get hashCode => Object.hash(hour, minute);

  @override
  String toString() {
    String addLeadingZeroIfNeeded(int value) {
      if (value < 10) {
        return '0$value';
      }
      return value.toString();
    }

    final String hourLabel = addLeadingZeroIfNeeded(hour);
    final String minuteLabel = addLeadingZeroIfNeeded(minute);

    return '$TimeOfDay($hourLabel:$minuteLabel)';
  }
}

class ConditionalCompare extends BenchmarkBase {
  ConditionalCompare() : super('Conditional compare');

  late List<TimeOfDay> data;

  @override
  void run() {
    data.sort((a, b) {
      final int hourComparison = a.hour.compareTo(b.hour);
      return hourComparison == 0 ? a.minute.compareTo(b.minute) : hourComparison;
    });
  }

  @override
  void setup() async {
    data = [
      for (int i = 0; i < 50; i++)
        for (int h = 0; h < 24; h++)
          for (int m = 59; m >= 0; m--)
            TimeOfDay(hour: h, minute: m),
    ];
  }

  @override
  void teardown() {}
}

class MathCompare extends BenchmarkBase {
  MathCompare() : super('Math compare');

  late List<TimeOfDay> data;

  @override
  void run() {
    data.sort((a, b) => 60 * a.hour.compareTo(b.hour) + a.minute.compareTo(b.minute));
  }

  @override
  void setup() async {
    data = [
      for (int i = 0; i < 50; i++)
        for (int h = 0; h < 24; h++)
          for (int m = 59; m >= 0; m--)
            TimeOfDay(hour: h, minute: m),
    ];
  }

  @override
  void teardown() {}
}

void main() {
  ConditionalCompare().report();
  MathCompare().report();
}
```

</p>
</details>

Reply:
Thanks for running the benchmark!
Perhaps 0.01 seconds isn't a huge deal, but I'm definitely in favor of performance improvements of any kind 🙂👍

Reply:
And I agree with your point regarding the purpose of `compareTo`. Resolving!